### PR TITLE
Tool/elprep merge

### LINKED ISSUE #1487: new module: Elprep
I think it would be good to have a module for [Elprep](https://bioconda.github.io/recipes/elprep/README.html)
This tool offer integrated bam sort/markdup and variant calling.

- [x] This module does not exist yet with the [`nf-core modules list`](https://github.com/nf-core/tools#list-modules) command
- [x] There is no [open pull request](https://github.com/nf-core/modules/pulls) for this module
- [x] There is no [open issue](https://github.com/nf-core/modules/issues) for this module
- [x] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by matthdsm**
## PR checklist

Depends on #1533 
Closes #1487

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by matthdsm** [comment]
Local tests seem to work fine. No idea why github fails...

**Comment by matthdsm** [comment]
The raw log show the input from `elprep split` is also output, after which `merge` tries to merge it too, but the file doesn`t exist

```
2022-04-22T09:53:21.5882352Z [38;5;238m  40[0m   [38;5;238m│[0m [38;5;231m  2022/04/22 09:53:11 Executing command:[0m
2022-04-22T09:53:21.5882881Z [38;5;238m  41[0m   [38;5;238m│[0m [38;5;231m   elprep merge input test.bam --nr-of-threads 2[0m
2022-04-22T09:53:21.5883505Z [38;5;238m  42[0m   [38;5;238m│[0m [38;5;231m  2022/04/22 09:53:11 open input/test-spread.bam: no such file or directory[0m
```

File: modules/elprep/merge/main.nf [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    tuple val(meta), path("output/**.{bam,sam}")    , emit: bam
+    path "versions.yml"                             , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args        = task.ext.args ?: ''
+    def prefix      = task.ext.prefix ?: "${meta.id}"
+    def suffix      = args.contains("--output-type sam") ? "sam" : "bam"
```
Comment (jfy133) [inline modules/elprep/merge/main.nf:23]: THAT IS CLEVER!

File: modules/elprep/merge/main.nf [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    def single_end  = meta.single_end ? " --single-end" : ""
+
+    """
+    # create directory and move all input so elprep can find and merge them before splitting
+    mkdir input
+    mv ${bam} input/
+
+    elprep merge \\
+        input \\
```
Comment (jfy133) [inline modules/elprep/merge/main.nf:32]:
```suggestion
        input/ \\
```

**Review by jfy133 — APPROVED** [review]

**Pull request merged**